### PR TITLE
feat: [PL-22538]: exposed GetLatestCommitOnFile as SCM API

### PR DESCRIPTION
**Pull request by mohitg0795**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30753)

### CODE REVIEW

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30753#-:-Mvvm_WY124Bn5vcmeZb:bnfp4nl)* status: :shipit: complete! ✓

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30753)* status: :shipit: complete! ✓

File: product/ci/scm/file/file.go [hunk 1]
```diff
@@ -416,11 +416,17 @@ func parseCrudResponse(ctx context.Context, client *scm.Client, body io.Reader,
...
 		// We try to find out the latest commit on the file, which is most-likely the commit done by SCM itself
 		// It works on best-effort basis
-		commit, err := git.GetLatestCommitOnFile(ctx, p, request.Slug, request.Branch, request.FilePath, log)
+		getLatestCommitRequest := &pb.GetLatestCommitOnFileRequest {
+			Slug: request.Slug,
+			Branch: request.Branch,
+			Provider: &p,
+			FilePath: request.FilePath,
+		}
+		getLatestCommitResponse, err := git.GetLatestCommitOnFile(ctx, getLatestCommitRequest, log)
```
[the remark this answers is not in the document]
Comment (shubham149) [reply product/ci/scm/file/file.go:425]: check this article for naming convention in golang: https://betterprogramming.pub/naming-conventions-in-go-short-but-descriptive-1fa7c6d2f32a#:~:text=The%20convention%20in%20Go%20is,can%20safely%20stick%20to%20mixedCaps%20.
Compared to java, it is expected to use shorter names even 1 char is allowed if context is small. e.g. you can rename getLatestCommitRequest in L419 to req.

**Comment by mohitg0795** [comment]
___
*[product/ci/scm/file/file.go, line 425 at r1](https://reviewable.io/reviews/harness/harness-core/30753#-MvttErX9hPLbEmeapm1-r1-425:-Mw1pKcV9P4g7T7w_pES:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/90cc061719d638740b5779188395168de99e8b0b/product/ci/scm/file/file.go#L425)):*
<details><summary><i>Previously, shubham149 (Shubham Agrawal) wrote…</i></summary><blockquote>

check this article for naming convention in golang: https://betterprogramming.pub/naming-conventions-in-go-short-but-descriptive-1fa7c6d2f32a#:~:text=The%20convention%20in%20Go%20is,can%20safely%20stick%20to%20mixedCaps%20.
Compared to java, it is expected to use shorter names even 1 char is allowed if context is small. e.g. you can rename getLatestCommitRequest in L419 to req.
</blockquote></details>

done

**Review by harshjain12 — DISMISSED** [review]
:lgtm:

> Reviewed 4 of 6 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30753#-:-MwQTLyI1nj-uXbuUbhG:bnfp4nl)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by mohitg0795** [comment]
trigger messagecheck

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30753#-:-MwQa2OJEbdEKNDjlOgS:bnfp4nl)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 4 of 6 files at r1, 2 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30753)* status: 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30753)* status: 1. unresolved discussions.

**Pull request merged**